src/logic/taskcluster.ts

```diff
@@ -0,0 +1,43 @@
+// This file contains logic for the Taskcluster Third-Party Login
+
+import { prodFirefoxRootUrl, getRootUrl } from '../common/constants';
+import { UserCredentials } from '../types/types';
+import { generateNonce } from '../utils/helpers';
+
+// For testing purposes
+const treeherderClientId = `treeherder-localhost-5000-client`;
+const treeherderRedirectURI = `http://localhost:5000/taskcluster-auth`;
+
+const getAuthCode = (defaultRootUrl: string) => {
+  const nonce = generateNonce();
+  localStorage.setItem('requestState', nonce);
+  localStorage.setItem('tcRootUrl', defaultRootUrl);
```

Comment:
My understanding is that we store it so that we can use it in the callback, especially so that we can use the staging server if we initiated the request to the staging server. Is that right? Can you add a small comment for that?